src/components/ShowList.jsx

```diff
@@ -50,26 +50,36 @@ function ShowList() {
               aria-label="clear filter text"
               className="clear-button"
             >
-              <i class="fas fa-times"></i>
+              <i className="fas fa-times"></i>
             </button>
           )}
+          {actionMessage ? (
+            <p className="success-message">{actionMessage}</p>
+          ) : null}
           <ul>
             {filter
               ? value.docs
-                  .map((doc) => ({ id: doc.id, ...doc.data() }))
                   .filter((item) =>
-                    item.name.toLowerCase().includes(filter.toLowerCase()),
+                    item
+                      .data()
+                      .name.toLowerCase()
+                      .includes(filter.toLowerCase()),
                   )
                   .map((item) => (
                     <Item
                       key={item.id}
-                      id={doc.id}
-                      lastPurchasedDate={item.lastPurchasedDate}
-                      name={item.name}
+                      setActionMessage={setActionMessage}
```

Comment:
Instead of passing the `setActionMessage` function from the useState hook, consider passing it a function that accepts a string and can handle the timeout. Right now, where this is being called, the caller also has to remember to remove the action message.

```js
const displayMessage = (message) => {
  setActionMessage(message)
  setTimeout(() => {
    setActionMessage(null);
  }, 10000);
}
```

This will keep all the logic together so the caller doesn't have to remember to also unset the action message.

Just pass it in here and the one below as:
```jsx
setActionMessage={displayMessage}
```